File: Neon-Java/src/com/islandstudio/neon/Stable/primary/iFolder/FolderList.java
package com.islandstudio.neon.stable.primary.iFolder;

import com.islandstudio.neon.stable.primary.iConstructor.IConstructor;
import com.islandstudio.neon.stable.secondary.iServerConstantProcessor.IServerConstantProcessor;

import java.io.File;

public enum FolderList {
    /* All the required folders will be here */
    DATA_FOLDER(IConstructor.getPlugin().getDataFolder()),
    MODE_FOLDER(new File(FolderList.DATA_FOLDER.folder, IFolder.getVersion() + "/" + IServerConstantProcessor.getServerMode())),

    SERVER_CONFIGURATION(new File(FolderList.MODE_FOLDER.folder, "/Server_Configuration")),
    PLAYER_DATA(new File(FolderList.MODE_FOLDER.folder, "/Server_Data/Player_Data")),
    IWAYPOINTS(new File(FolderList.MODE_FOLDER.folder, "/Server_Data/iWaypoints")),
    IEXPERIMENTAL(new File(FolderList.MODE_FOLDER.folder, "/Server_Data/iExperimental"));

    private final File folder;

    FolderList(File folder) {
        this.folder = folder;
    }

    public File getFolder() {
        return folder;
    }
}

File: Neon-Kotlin/src/main/com/islandstudio/neon/stable/primary/nFolder/FolderList.kt
package com.islandstudio.neon.stable.primary.nFolder

import com.islandstudio.neon.stable.primary.nServerConstantEventProcessor.NServerConstantProcessor
import java.io.File

enum class FolderList(val folder: File) {
    /* All the required folders will be here */
    /* Stable */
    MODE_FOLDER(File(NFolder.getDataFolder(), "${NFolder.getVersion()}/${NServerConstantProcessor.getMode()}")),
    NSERVERFEATURES_FOLDER(File(MODE_FOLDER.folder, "nServerFeatures")),
    NEXPERIMENTAL_FOLDER(File(NSERVERFEATURES_FOLDER.folder, "nExperimental")),
    NPROFILE(File(MODE_FOLDER.folder, "nProfile")),
    NWAYPOINTS_FOLDER(File(NSERVERFEATURES_FOLDER.folder, "nWaypoints")),

    /* Experimental */
    NFIREWORKS(File(NEXPERIMENTAL_FOLDER.folder, "nFireworks")),
    NFIREWORKS_IMAGES(File(NFIREWORKS.folder, "images")),
    NFIREWORKS_PARTICLE_DATA(File(NFIREWORKS.folder, "particle_data"))
}